core/modules/xrdsvc/StreamBuffer.cc

```diff
@@ -38,17 +38,42 @@ LOG_LOGGER _log = LOG_GET("lsst.qserv.xrdsvc.StreamBuffer");
 }
 
 
+using namespace std;
+
+
 namespace lsst {
 namespace qserv {
 namespace xrdsvc {
 
-std::atomic<size_t> StreamBuffer::_totalBytes(0);
+
+atomic<int64_t> StreamBuffer::_maxTotalBytes{40'000'000'000};
+atomic<int64_t> StreamBuffer::_totalBytes(0);
+mutex StreamBuffer::_createMtx;
+condition_variable StreamBuffer::_createCv;
+
+
+static void StreamBuffer::setMaxTotalBytes(int64_t maxBytes) {
+    LOGS(_log, LOG_LVL_INFO, "StreamBuffer::setMaxTotalBytes maxBytes=" << maxBytes);
+    if (maxBytes < 0) {
```

Comment:
You wouldn't have this check should the `maxBytes` be defined as an unsigned integer.

Reply:
Perhaps, but if someone puts a -1 in the config file, what happens? ConfigStore only has getInt(), nothing for unsigned. Making the check is pretty trivial.